### PR TITLE
oneway support for pedestrian access and foot restrictions

### PR DESCRIPTION
**Pull request by gknisely**
# Issue

Allows support for oneway pedestrian ways.

### Before.  Via oneway.  (Ignores oneway exit for a ride)
Screenshot from 2021-06-14 21-52-22

### After.  Via oneway.
Screenshot from 2021-06-14 21-53-35

### Before.  Ending on onway exit for a ride.
Screenshot from 2021-06-14 21-52-08

### After.  Ending on onway exit for a ride.
Screenshot from 2021-06-14 21-53-19

### Before.  Oneway footpath
Screenshot from 2021-06-14 21-51-57

### After.  Oneway footpath
Screenshot from 2021-06-14 21-53-09

### Before.  Foot restriction (enter only)  Restriction ignored.
Screenshot from 2021-06-14 21-51-20

### After.  Foot restriction (enter only)
Screenshot from 2021-06-14 21-52-40

## Tasklist

 - [x] Add tests
 - [x] Update the [changelog](CHANGELOG.md)
 - [x] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: lua/graph.lua [hunk 1]
```diff
@@ -1173,10 +1197,15 @@ function filter_tags_generic(kv)
       kv["moped_backward"] = kv["moped_forward"]
     end
 
-    if (kv["motorcycle_backward"] == "false" and (kv["oneway:motorcycle"] == nil or oneway[kv["oneway:motorcycle"]] == false or kv["oneway:motorcycle"] == "no")) then
```
Comment (gknisely) [inline lua/graph.lua:1176]: Small update.  Found a bug via the new gurka test

File: lua/graph.lua [hunk 2]
```diff
@@ -1237,6 +1270,24 @@ function filter_tags_generic(kv)
     kv["bike_backward"] = forwards
   end
 
+  if (kv["oneway:moped"] == "-1" or kv["oneway:mofa"] == "-1") then
```
Comment (gknisely) [inline lua/graph.lua:1273]: More issues found via the gurka test from lines 1273 to 1283

File: src/mjolnir/directededgebuilder.cc [hunk 3]
```diff
@@ -134,14 +134,24 @@ DirectedEdgeBuilder::DirectedEdgeBuilder(const OSMWay& way,
...
+  }
+  if ((way.pedestrian_forward() && !forward) || (way.pedestrian_backward() && forward)) {
     reverse_access |= kPedestrianAccess;
   }
-  if (way.use() != Use::kSteps &&
-      ((way.wheelchair_tag() && way.wheelchair()) || (!way.wheelchair_tag() && way.pedestrian()))) {
-    forward_access |= kWheelchairAccess;
-    reverse_access |= kWheelchairAccess;
+  if (way.use() != Use::kSteps) {
+    if (way.wheelchair_tag() && way.wheelchair()) {
```
Comment (gknisely) [inline src/mjolnir/directededgebuilder.cc:144]: Wheelchair routing and logic needs to be finished.  This was started a long time ago but never finished.

File: src/sif/pedestriancost.cc [hunk 4]
```diff
@@ -551,14 +551,17 @@ bool PedestrianCost::Allowed(const baldr::DirectedEdge* edge,
                              const uint64_t current_time,
                              const uint32_t tz_index,
                              uint8_t& restriction_idx) const {
-  if (!IsAccessible(edge) || (edge->surface() > minimal_allowed_surface_) || edge->is_shortcut() ||
```
Comment (gknisely) [inline src/sif/pedestriancost.cc:554]: Added restrictions check

**Review by merkispavel — APPROVED** [review]
Nice! The only thing is if someone could review the lua because I didn't

File: valhalla/mjolnir/osmway.h [hunk 5]
```diff
@@ -1689,7 +1707,7 @@ struct OSMWay {
   uint16_t use_sidepath_ : 1;
   uint16_t bike_forward_ : 1;
   uint16_t bike_backward_ : 1;
-  uint16_t spare1_ : 4;
+  uint16_t spare3_ : 4;
```
Comment (dgearhart) [inline valhalla/mjolnir/osmway.h:1710]: nit: why spare3?

**Review by dgearhart — DISMISSED** [review]
other than the nit - lgtm

**Review by dgearhart — APPROVED** [review]
:ship:

**Pull request merged**